src/interface/src/app/services/jwt-interceptor.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { catchError, Observable, switchMap } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class JwtInterceptor implements HttpInterceptor {
+  constructor(private auth: AuthService) {}
```

Comment:
I think we're using authService everywhere else, so maybe for consistency we should change this?